src/util/records.py

```diff
@@ -749,6 +749,14 @@ def rename_xmlid(cr, old, new, noupdate=None, on_collision="fail"):
                 # Don't change the view keys inconditionally to avoid changing unrelated views.
                 cr.execute("UPDATE ir_ui_view SET key = %s WHERE key = %s", [new, old])
 
+        if model == "ir.ui.menu" and column_exists(cr, "res_users_settings", "homemenu_config"):
+            query = """
+                UPDATE res_users_settings
+                   SET homemenu_config = regex_replace(homemenu_config, %(old)s, %(new)s)
+                 WHERE homemenu_config ~ %(old)s
+            """
+            cr.execute(query, {"old": r"\y{}\y".format(re.escape(old)), "new": new})
```

Comment:
Since we know it's a JSON encoded string array, can't we just use `replace`?
```suggestion
                   SET homemenu_config = replace(homemenu_config, %(old)s, %(new)s)
                 WHERE homemenu_config LIKE '%%' || %(old)s '%%'
            """
            cr.execute(query, {"old": '"{}"'.format(old), "new": '"{}"'.format(new)})
```

Reply:
It's double-encoded JSON.
So, it currently looks like 
```
 "[\"knowledge.knowledge_menu_root\",\"calendar.mail_menu_calendar\",\"appointment.main_menu_appointments\",\"project_todo.menu_todo_todos\",\"contacts.menu_contacts\",\"crm.crm_menu_root\",\"sale.sale_menu_root\",\"spreadsheet_dashboard.spreadsheet_dashboard_menu_root\",\"sale_subscription.menu_sale_subscription_root\",\"project.menu_main_pm\",\"hr_timesheet.timesheet_menu_root\",\"planning.planning_menu_root\",\"website.menu_website_configuration\",\"event.event_main_menu\",\"purchase.menu_purchase_root\",\"maintenance.menu_maintenance_title\",\"sign.menu_document\",\"hr.menu_hr_root\",\"hr_appraisal.menu_hr_appraisal_root\",\"mail.menu_root_discuss\",\"hr_referral.menu_hr_referral_root\",\"hr_holidays.menu_hr_holidays_root\",\"hr_expense.menu_hr_expense_root\"]"
```

If we use `replace` with the quote, we will have to adapt this code once this double-encoding is fixed.

Reply:
🤦🏼